x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/SqlFunctionRegistryTests.java

```diff
@@ -0,0 +1,130 @@
+
+import static org.elasticsearch.xpack.ql.TestUtils.randomConfiguration;
+import static org.elasticsearch.xpack.ql.expression.function.FunctionRegistry.def;
+import static org.elasticsearch.xpack.ql.expression.function.FunctionRegistryTests.uf;
+import static org.elasticsearch.xpack.ql.expression.function.FunctionResolutionStrategy.DEFAULT;
+import static org.elasticsearch.xpack.sql.expression.function.SqlFunctionResolution.DISTINCT;
+import static org.elasticsearch.xpack.sql.expression.function.SqlFunctionResolution.EXTRACT;
+import static org.hamcrest.Matchers.endsWith;
+import static org.mockito.Mockito.mock;
+
+public class SqlFunctionRegistryTests extends ESTestCase {
+
+    public void testNoArgFunction() {
+        UnresolvedFunction ur = uf(DEFAULT);
+        FunctionRegistry r = new FunctionRegistry(def(DummyFunction.class, DummyFunction::new, "DUMMY_FUNCTION"));
+        FunctionDefinition def = r.resolveFunction(ur.name());
+        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
+
+        // Distinct isn't supported
+        ParsingException e = expectThrows(ParsingException.class, () ->
+            uf(DISTINCT).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("does not support DISTINCT yet it was specified"));
+
+        // Any children aren't supported
+        e = expectThrows(ParsingException.class, () ->
+            uf(DEFAULT, mock(Expression.class)).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("expects no arguments"));
+    }
+
+    public void testUnaryFunction() {
+        UnresolvedFunction ur = uf(DEFAULT, mock(Expression.class));
+        FunctionRegistry r = new FunctionRegistry(def(DummyFunction.class, (Source l, Expression e) -> {
+            assertSame(e, ur.children().get(0));
+            return new DummyFunction(l);
+        }, "DUMMY_FUNCTION"));
+        FunctionDefinition def = r.resolveFunction(ur.name());
+        assertFalse(def.extractViable());
+        assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
+
+        // Distinct isn't supported
+        ParsingException e = expectThrows(ParsingException.class, () ->
+            uf(DISTINCT, mock(Expression.class)).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("does not support DISTINCT yet it was specified"));
+
+        // No children aren't supported
+        e = expectThrows(ParsingException.class, () ->
+            uf(DEFAULT).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("expects exactly one argument"));
+
+        // Multiple children aren't supported
+        e = expectThrows(ParsingException.class, () ->
+            uf(DEFAULT, mock(Expression.class), mock(Expression.class)).buildResolved(randomConfiguration(), def));
+        assertThat(e.getMessage(), endsWith("expects exactly one argument"));
```

Comment:
These tests are duplicates of the method with the same name in `FunctionRegistryTests` in QL.   Why duplicating these?

Reply:
The methods in the base class don't contain all the strategies - the first lines are code are duplicated (the creation is needed, plus the two assertions) but the rest in the subclass are not.

Reply:
You could keep only the initialization part and the DISTINCT test, no?

Reply:
I'll look further at eliminating the extra assertions.